### PR TITLE
clarify the error relaying of EADDRINUSE for uv_tcp_bind

### PR DESCRIPTION
**Pull request by deleisha**
doc is updated to give the proper behaviour

### CODE REVIEW

File: docs/src/tcp.rst [hunk 1]
```diff
@@ -75,11 +75,11 @@ API
     Bind the handle to an address and port. `addr` should point to an
     initialized ``struct sockaddr_in`` or ``struct sockaddr_in6``.
 
-    When the port is already taken, you can expect to see an ``UV_EADDRINUSE``
-    error from either :c:func:`uv_tcp_bind`, :c:func:`uv_listen` or
-    :c:func:`uv_tcp_connect`. That is, a successful call to this function does
-    not guarantee that the call to :c:func:`uv_listen` or :c:func:`uv_tcp_connect`
-    will succeed as well.
+    When the port is already taken, the error is relayed and you can expect to see an 
+    ``UV_EADDRINUSE`` error from function such as :c:func:`uv_listen` or
```
Comment (bnoordhuis) [inline docs/src/tcp.rst]: I think 'deferred' is a better choice of words than 'relayed'.   Also, why the introduction of 'function such as'?

**Comment by bnoordhuis** [comment]
Can you explain in the commit log in a few words why you think this makes things more clear?  I'm a bit on the fence, I'm not sure it adds much.

File: docs/src/tcp.rst (hunk 1 above)
Comment (deleisha) [inline docs/src/tcp.rst]: Yes, deferred is a better choice. 
preliminary code scan shows that delayed_error is being returned at
https://github.com/deleisha/libuv/blob/v1.x/src/unix/tcp.c#L212
https://github.com/deleisha/libuv/blob/v1.x/src/unix/tcp.c#L234

Hence, My thinking was that function like `uv_tcp_getpeername` and `uv_tcp_getsockname`.

**Comment by deleisha** [comment]
sure, I can update the commit log once your other comments are taken care. `Why this PR?`  is because my understanding is that `uv_tcp_bind` will not  return EADDRINUSE.

**Comment by deleisha** [comment]
My bad, I have mixed two things here. Let me send two different PR and close this for now

**Pull request closed without merging**